src/workerd/jsg/iterator.h

```diff
@@ -276,13 +281,19 @@ class AsyncGenerator final {
     kj::Maybe<NextSignature> maybeNext;
     kj::Maybe<ReturnSignature> maybeReturn;
     kj::Maybe<ThrowSignature> maybeThrow;
+    // Per GetMethod spec, if property exists but is not callable, we should throw TypeError.
+    // We track this state to defer the error to when return_() is actually called.
+    bool returnExistsButNotCallable = false;
 
     template <typename TypeWrapper>
     Active(Lock& js, JsObject object, TypeWrapper*)
         : maybeNext(tryGetGeneratorFunction<NextSignature, TypeWrapper>(js, object, "next"_kj)),
           maybeReturn(
               tryGetGeneratorFunction<ReturnSignature, TypeWrapper>(js, object, "return"_kj)),
           maybeThrow(tryGetGeneratorFunction<ThrowSignature, TypeWrapper>(js, object, "throw"_kj)) {
+      // Check if return property exists but isn't callable (per GetMethod spec)
+      returnExistsButNotCallable =
+          maybeReturn == kj::none && !object.get(js, "return"_kj).isNullOrUndefined();
```

Comment:
This is a bit non-obvious. The variable is called `...NotCallable` but it's being set with a call to `isNullOrUndefined`. This could use a bit more explanation to be clearer explaining that if `return` is not a function, `maybeReturn` will be `kj::none` and that the additional check here is just an indicator of whether the property actually existed or not (using a `has(...)` might be better if we keep this check to avoid the additional handle creation?).

Also, the fact that the `return` property is being looked up twice is rather unfortunate, especially given that the lookup could have side effects in some edge cases and subsequent accesses are not necessarily always guaranteed to yield the same value.